### DISCUSSION TITLE
右手用基盤で設定したLEDパターンで点灯しない

### DISCUSSION THREAD

**Discussion by josh3018** [Yowkees/keyball#662] [Q&A]
keyball44を購入・自作させていただき、一ヶ月ほどになります。
下記の不具合があり、対策のためのご助言をいただけないでしょうか？

【症状概要】
右手基盤だけ、設定したLEDパターンで点灯するときとしないときがある。
設定と異なるパターンで点灯するときは、特定の色で輝度100で点灯する。（添付写真1枚目、青色）
一度、キーボード操作でLEDを消し、再度点灯すれば、設定したパターンで点灯する。（添付写真2枚目、紫色）

![PXL_20240921_050742916](https://github.com/user-attachments/assets/c0c7dcbe-aa91-40ee-9d76-cbf5100c57aa)
![PXL_20240921_050819464](https://github.com/user-attachments/assets/edf158a0-0e40-41aa-bd80-cad90175f9a2)

【症状詳細】
- 机の上に直置きやケースに入れた状態で電源を入れると、上記の症状が10回中3~5回ほど発生する
- 基盤を手に持って、基盤裏の部品が何も接触していない状態で電源を入れると、上記症状が出ない
- Promicroを左右入れ替えても、この症状が出るのは右手基盤のみ
- TSRケーブルの接続有無で症状に変化なし

【基盤裏写真】
はんだ付けの状況から、何か原因などわかりますでしょうか？
右手基盤ではLEDはんだ付け時に何個かのLEDで点灯不良があり、はんだ付けし直した点が、左手基盤と異なります。
![PXL_20240921_050333755~2](https://github.com/user-attachments/assets/9102bc40-88b4-464f-a409-cd508d03950f)

通常使用に問題はないのですが、後学のために、このような場合の原因調査方法・解決方針をご教示いただけないでしょうか？

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

ちょっと経験したことのない症状ですが、ProMicroの電流不足が関係しているように感じました。
LED設定を紫色一色に設定しているとお見受けしましたが、その他の色だと再現するでしょうか？

白っぽい色で輝度を高くすると、ProMicroの１ピンあたりの電流限界に達する場合があります。

以上、お手数ですがご確認いただければ幸いです。

**Reply by josh3018** [reply]
@Yowkees 
お忙しいところ、誠にありがとうございます。

はい、全ての色味を見れてはおりませんが、他の色でも再現いたします。輝度は40で設定していますが、それでも電流不足等の可能性があるものでしょうか？

基盤裏の部品を何かに接触させると不備が起きるといえ点も、たまたまそう見えているだけで、そこで原因を切り分けるのはなかなか難しそうでしょうか。
